cargo-shuttle/src/init.rs

```diff
@@ -108,3 +343,100 @@ fn create_gitignore_file(path: &Path) -> Result<()> {
 
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    macro_rules! test_gen_from {
+        // Two string literals, the second of which becomes an `Option`.
+        ($auto_path:literal, $subfolder:literal) => {
+            GenerateFrom::from(TemplateLocation {
+                auto_path: $auto_path.to_owned(),
+                subfolder: Some($subfolder.to_owned()),
+            })
+        };
+
+        // A string literal and an expression (`None`).
+        ($auto_path:literal, $subfolder:expr) => {
+            GenerateFrom::from(TemplateLocation {
+                auto_path: $auto_path.to_owned(),
+                subfolder: $subfolder,
+            })
+        };
+    }
+
+    #[test]
+    fn parse_valid_template_location() {
```

Comment:
We are testing that these do parse, but not if the result is equal to the expected.
Adding that to all of these seems excessive, but adding it for a few of these can do. Good for verifying the path concatenation, for example.